test/e2e/run.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash -e
+
+export NOTATION_E2E_BINARY_PATH=$(realpath $1)
+if [ ! -f "$NOTATION_E2E_BINARY_PATH" ];then
+    echo "run.sh <notation-binary-path>"
+    exit 1
+fi
+
+# install dependency
+go install -mod=mod github.com/onsi/ginkgo/v2/ginkgo
+
+# set environment variable for E2E testing
+REG_HOST=localhost
+REG_PORT=5000
+ZOT_CONTAINER_NAME=zot
```

Comment:
Can we make test generic and not tightly coupled to zot?

Reply:
Refactor the run.sh to make it easier for multi-registry testing.